src/model/OWL.plt

```diff
@@ -0,0 +1,547 @@
+  assert_false(is_transitive_property(test:'DoesNotExists')),
+  % Argument is unbound
+  assert_true(is_transitive_property(_)),
+  assert_true(once((is_transitive_property(A), test:'r'=A))).
+
+test('is_symmetric_property') :-
+  assert_false(is_symmetric_property(test:'p')),
+  assert_true(is_symmetric_property(test:'s')),
+  assert_false(is_symmetric_property(test:'r')),
+  assert_false(is_symmetric_property(test:'DoesNotExists')),
+  % Argument is unbound
+  assert_true(is_symmetric_property(_)),
+  assert_true(once((is_symmetric_property(A), test:'s'=A))).
+
+test('has_inverse_property') :-
+  assert_true(has_inverse_property(test:'r',test:'rInv')),
+  assert_true(has_inverse_property(test:'rInv2',test:'r')),
+  % Left argument is unbound
+  assert_true(has_inverse_property(test:'r',_)),
+  % Right argument is unbound
+  assert_true(has_inverse_property(_,test:'rInv')),
+  % Both arguments are unbound
+  assert_true(has_inverse_property(_,_)),
+  % Negative Case
+  assert_false(has_inverse_property(test:'p',test:'r')).
+
+test('has_property_chain') :-
+  % Left argument is unbound
+  assert_true(
+    has_property_chain(_,[test:'t1',test:'t2',test:'t3'])
+  ),
+  assert_true(
+    once((
+      has_property_chain(A,[test:'t1',test:'t2',test:'t3']), 
+      A=test:'t'
+    ))
+  ), 
+  % Right argument is unbound
+  assert_true(has_property_chain(test:'t',_)),
+  assert_true(
+    once((has_property_chain(test:'t',B), 
+      [test:'t1',test:'t2',test:'t3']=B))
+  ),
+  % Both arguments are unbound
+  assert_true(has_property_chain(_,_)),!.
+
+test('has_property_chain with unbound arguments') :-
+  % Left argument is unbound
+  assert_true(
+    has_property_chain(_,[test:'t1',test:'t2',test:'t3'])
+  ),
+  assert_true(
+    once((
+      has_property_chain(A,[test:'t1',test:'t2',test:'t3']), 
+      A=test:'t'
+    ))
+  ), 
+  % Right argument is unbound
+  assert_true(has_property_chain(test:'t',_)),
+  assert_true(
+    once((has_property_chain(test:'t',B), 
+      [test:'t1',test:'t2',test:'t3']=B))
+  ).
+
+%test('has_disjoint_class1', 
+%    [ blocked('The transitive(subclass_of(B,Disjoint) call in unify_disjoint of OWL.pl seems to not return the transitive closure of all subclasses') ]) :-
+test('has_disjoint_class1') :-
+  assert_true(
+    has_disjoint_class(test:'DisjClsChain1',test:'DisjClsChain2')
+  ),
+  assert_true(
+    has_disjoint_class(test:'DisjClsChain2',test:'DisjClsChain1')
+  ),
+  assert_true(
+    has_disjoint_class(test:'DisjClsChain2',test:'DisjClsChain3')
+  ),
+  assert_false(
+    has_disjoint_class(test:'DisjClsChain1',test:'DisjClsChain3')
+  ),
+  % Negative Case
+  assert_false(has_disjoint_class(test:'A',test:'B')).
+
+test('has_disjoint_class1 with unbound arguments') :-
+  % Left argument is unbound
+  assert_true(has_disjoint_class(_,test:'DisjClsChain2')),
+  assert_true((
+    findall(A, has_disjoint_class(A,test:'DisjClsChain2'), AList), 
+    member(test:'DisjClsChain1', AList)
```

Comment:
better use `memberchk/2` as it is semi-determistic

https://www.swi-prolog.org/pldoc/doc_for?object=memberchk/2